x/rollup/keeper/msg_server.go

```diff
@@ -83,3 +86,70 @@ func (k *Keeper) InitiateWithdrawal(
 
 	return &types.MsgInitiateWithdrawalResponse{}, nil
 }
+
+func (k *Keeper) InitiateFeeWithdrawal(
+	goCtx context.Context,
+	_ *types.MsgInitiateFeeWithdrawal,
+) (*types.MsgInitiateFeeWithdrawalResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// TODO: make minWithdrawalAmount and l1recipientAddr configurable once a genesis state is added
+	const (
+		minWithdrawalAmount   = 100
+		l1recipientAddr       = "0x63d93aC6FA6B4021527e967ac3Eb29F2B3E52B96"
+		feeWithdrawalGasLimit = 400_000
+	)
+
+	feeCollectorAddr := k.accountkeeper.GetModuleAddress(authtypes.FeeCollectorName)
+	if feeCollectorAddr == nil {
+		ctx.Logger().Error("Failed to get fee collector address")
+		return nil, types.WrapError(types.ErrInitiateFeeWithdrawal, "failed to get fee collector address")
```

Comment:
Is the standard pattern to log and then return an error? I feel like the error message should be logged further up.

Reply:
Yeah I agree that this pattern is pretty confusing and not too helpful, especially since we haven't gotten any user feedback on debugging Monomer L2s yet.

I think the initial `ctx.Logger()...` pattern before the SDK error was inherited code and I decided to just keep it going. I'm leaning towards removing all of the `ctx.Logger()...` calls in `x/rollup` until we do some usability testing, is that alright with you?

Reply:
Let's ensure the errors get logged somewhere and remove if they do get logged